backend/src/test/java/ddangkong/domain/balance/room/RoomTest.java

```diff
@@ -41,4 +43,118 @@ class 다음_라운드로_이동 {
+            // given
+            int currentRound = 2;
+            Room room = new Room(FIXED_TOTAL_ROUND, currentRound, FIXED_TIME_LIMIT, FIXED_STATUS);
+            int myRound = 2;
+
+            // when
+            boolean actual = room.isMyRoundFinished(myRound);
+
+            // then
+            assertThat(actual).isFalse();
+        }
+
+        @Test
+        void 나의_라운드가_방의_시작_라운드보다_작으면_예외가_발생한다() {
+            // given
+            Room room = new Room(FIXED_TOTAL_ROUND, 1, FIXED_TIME_LIMIT, FIXED_STATUS);
+            int invalidMyRound = 0;
+
+            // when & then
+            assertThatThrownBy(() -> room.isMyRoundFinished(invalidMyRound))
+                    .isExactlyInstanceOf(BadRequestException.class)
+                    .hasMessageContaining("startRound보다 크거나 같아야 합니다. startRound : 1, round : 0");
+        }
+
+        @Test
+        void 나의_라운드가_방의_현재_라운드보다_크면_예외가_발생한다() {
+            // given
+            int currentRound = 1;
+            Room room = new Room(FIXED_TOTAL_ROUND, currentRound, FIXED_TIME_LIMIT, FIXED_STATUS);
+            int invalidMyRound = 2;
+
+            // when & then
+            assertThatThrownBy(() -> room.isMyRoundFinished(invalidMyRound))
+                    .isExactlyInstanceOf(BadRequestException.class)
+                    .hasMessageContaining("currentRound보다 작거나 같아야 합니다. currentRound : 1, round : 2");
+        }
+
+        @Test
+        void 나의_라운드가_방의_현재_라운드보다_2이상_작으면_예외가_발생한다() {
+            // given
+            int currentRound = 4;
+            Room room = new Room(FIXED_TOTAL_ROUND, currentRound, FIXED_TIME_LIMIT, FIXED_STATUS);
+            int invalidMyRound = 2;
+
+            // when & then
+            assertThatThrownBy(() -> room.isMyRoundFinished(invalidMyRound))
+                    .isExactlyInstanceOf(BadRequestException.class)
+                    .hasMessageContaining("currentRound과 round의 차이는 1이하여야 합니다. currentRound : 4, round : 2");
+        }
+
+        @Test
+        void 현재_라운드와_전체_라운드가_같고_방_상태가_FINISH이면_방의_전체_라운드가_종료된_것이다() {
+            // given
+            RoomStatus status = RoomStatus.FINISH;
+            Room room = new Room(FIXED_TOTAL_ROUND, 5, FIXED_TIME_LIMIT, status);
+
+            // when
+            boolean actual = room.isAllRoundFinished();
+
+            // then
+            assertThat(actual).isTrue();
+        }
+
+        @Test
+        void 현재_라운드와_전체_라운드가_다르면_방의_전체_라운드가_종료되지_않은_것이다() {
+            // given
+            int currentRound = 3;
+            int totalRound = 5;
+            Room room = new Room(totalRound, currentRound, FIXED_TIME_LIMIT, FIXED_STATUS);
+
+            // when
+            boolean actual = room.isAllRoundFinished();
+
+            // then
+            assertThat(actual).isFalse();
+        }
+
+        @ParameterizedTest
+        @EnumSource(value = RoomStatus.class, names = {"READY", "PROGRESS"})
+        void 방_상태가_FINISH가_아니면_방의_전체_라운드가_종료되지_않은_것이다(RoomStatus status) {
```

Comment:
사소한 의견) `@CsvSource` 쓰면 굳이 value를 명시하지 않아도 될 것 같아요. ( `@CsvSource` 보다 `@EnumSource`가 더 좋은 이유 있다면 알려주세요. 저도 쓰게...)
```suggestion
        @ParameterizedTest
        @CsvSource({"READY", "PROGRESS"})
        void 방_상태가_FINISH가_아니면_방의_전체_라운드가_종료되지_않은_것이다(RoomStatus status) {
```

Reply:
`@CsvSource`도 enum을 넣을 수 있군요 처음 알았네요!

더 좋은 이유보다는 쓰임이 다르다라고 답할 수 있을 것 같아요 `@CsvSource`는 csv 형식을 따르고 파라미터로 여러 값을 줄 때 사용합니다
```java
@ParameterizedTest
@CsvSource({"1, true", "2, false"})
void test(int value1, boolean value2) {
    System.out.println("value1 = %d, value2 = %b".formatted(value1, value2));
}
```

<img width="255" alt="스크린샷 2024-08-02 오전 12 50 55" src="https://github.com/user-attachments/assets/cbf3e2c3-97a4-45ae-82d4-8381e0e3e78b">

객체나 Collection이라면 `@MethodSource`를 사용하곤 해요

`@EnumSource`는 enum에 정의된 상수를 테스트할 때 사용합니다
그리고 기본 모드는 include인데, exclude 모드로 변경하면 명시한 상수를 제외한 모든 상수를 테스트할 수 있어요
```java
@ParameterizedTest
@EnumSource(mode = Mode.EXCLUDE, names = {"FINISH"})
void 방_상태가_FINISH가_아니면_방의_전체_라운드가_종료되지_않은_것이다(RoomStatus status) {
    // ...
}
```
메서드명에 맞춰서 exclude로 변경하고 value는 파라미터에 타입 명시되어 있으면 생략 가능해서 제거했습니다 https://github.com/woowacourse-teams/2024-ddangkong/pull/102/commits/2338538817a2232f9fe2e42079c90b7795aeac37

Reply:
프린이 친절히 정리해 준 덕분에 제 코드에도 적용할 수 있겠네요! 고마워요!
저도 다음 코드에서 적용해보도록 할께요!!!!!!